modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/opentofu/opentofulocal/OpenTofuLocalExecutor.java

```diff
@@ -192,8 +197,13 @@ public void destroy() {
      */
     public String getTerraformState() {
         String state = null;
+        String path = workspace + File.separator + STATE_FILE_NAME;
         try {
-            File tfState = new File(workspace + File.separator + STATE_FILE_NAME);
+            Awaitility.await()
+                    .atMost(Duration.ofSeconds(AWAIT_AT_MOST))
```

Comment:
can you also please add `.pollDelay(0, TimeUnit.SECONDS) `?
other wise the check will start only after the first interval has passed.